core/src/main/java/org/apache/accumulo/core/metadata/schema/ExternalCompactionMetadata.java

```diff
@@ -129,10 +129,11 @@ private static class GSonData {
   public String toJson() {
     GSonData jData = new GSonData();
 
-    jData.inputs = jobFiles.stream().map(StoredTabletFile::getMetaUpdateDelete).collect(toList());
-    jData.nextFiles =
-        nextFiles.stream().map(StoredTabletFile::getMetaUpdateDelete).collect(toList());
-    jData.tmp = compactTmpName.getMetaInsert();
+    jData.inputs = jobFiles.stream()
+        .map(tabletFile -> tabletFile.getMetaUpdateDelete().getMetaString()).collect(toList());
+    jData.nextFiles = nextFiles.stream()
+        .map(tabletFile -> tabletFile.getMetaUpdateDelete().getMetaString()).collect(toList());
```

Comment:
This breaks when/if `getMetaString` changes to include the file range.

Reply:
Same as other comment, we could update it to be the file reference but no matter what it's going to still be a String value and will need to change in the future.